backend/src/main/java/com/google/minicrm/servlets/LeadsServlet.java

```diff
@@ -73,4 +79,95 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(leads));
   }
+
+  /**
+   * Updates the advertiser-modifiable fields (status and notes) of a lead specified by lead_id.
+   * Authentication required.
+   *
+   * Required parameters in the content body:
+   *  - lead_id : String uniquely identifying a lead
+   * Optional parameters to be updated:
+   *  - status : the status of the lead represented as a String
+   *  - notes : the notes of the lead represented as a String
+   * Accepted content-encodings:
+   *  - application/x-www-form-urlencoded
+   *  - application/json
+   *
+   * HTTP Response Status Codes:
+   * - 200 OK: lead edited successfully
+   * - 400 Bad Request: if the request does not contain the required fields or has an invalid status
+   * - 401 Unauthorized: if not logged in with Google
+   * - 404 Not Found: if the lead with the specified lead_id is not found
+   * - 415 Not Supported: if content body is not a valid type
+   *
+   * @param request  the HTTP Request. Expecting parameter form_id with the form_id to add
+   * @param response the HTTP Response
+   * @throws IOException if an input exception occurs with the response writer or reader
+   */
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    if (!UserAuthenticationUtil.isAuthenticated()) {
+      response.sendError(401, "Log in with Google to continue."); //401 Unauthorized
+      return;
+    }
+
+    //read the content body
+    String leadId;
+    String status;
+    String notes;
+    if (request.getContentType().contains("application/x-www-form-urlencoded")) {
+      leadId = request.getParameter("leadId");
+      status = request.getParameter("form_id");
+      notes = request.getParameter("form_name");
+    } else if (request.getContentType().contains("application/json")) {
```

Comment:
nit, put "request.getContentType()" as a variable since it is used more than once.

Reply:
will fix!